<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <h1>
        예상되는 에러를 try{code...} 영역에 배치하고, 만약 에러가 발생하면 catch(error){code...} 영역에서 처리하는 예외처리 프로세스
    </h1>
    <ul>
        <li>
            try : 에러 발생 가능성이 있는 코드를 실행하는 곳
        </li>
        <li>
            catch(err) : 에러 발생시 예외 처리를 하는 곳
        </li>
        <li>
            try~catch로 에외처리를 하면 프로세스를 멈추지 않고 지속시킬 수 있음.
        </li>
        <li>
            finally 영역 : 예외처리와 관계없이 무조건 코드를 실행하는 영역
        </li>
        <li>
            throw : 예외를 고의적으로 발생시키는 키워드 throw + 예외발생 후 처리할 코드
        </li>
    </ul>

    <script>
        console.log("프로세스 시작");
        try {
            alerttt("경고창 실행");
            console.log("경고창 실행 확인");
        } catch (error) {
            console.log("catch 영역");
            console.log(error.name);
            console.log(error.message);
            try {
                if (error.message.toString().length > 0) {
                    throw "throw error";
                }
            } catch (error2) {
                console.log(`error2 : ${error2}`);
            } finally {
                console.log("error2 finally...");
            }
        } finally {
            console.log("마지막 finally 영역");
        }
    </script>
</body>

</html>